pkg/objstore/oss/oss.go

```diff
@@ -0,0 +1,310 @@
+package oss
+
+import (
+	"context"
+	"fmt"
+	"io"
+	"math/rand"
+	"net/http"
+	"os"
+	"strings"
+	"testing"
+	"time"
+
+	alioss "github.com/aliyun/aliyun-oss-go-sdk/oss"
+	"github.com/go-kit/kit/log"
+	"github.com/improbable-eng/thanos/pkg/objstore"
+	"github.com/improbable-eng/thanos/pkg/runutil"
+	"github.com/pkg/errors"
+	yaml "gopkg.in/yaml.v2"
+)
+
+type Config struct {
+	EndPoint  string `yaml:"endpoint"`
+	Bucket    string `yaml:"bucket"`
+	SecretKey string `yaml:"secret_key"`
+	SecretId  string `yaml:"secret_id"`
+}
+
+type objectInfo struct {
+	key string
+	err error
+}
+
+type Bucket struct {
+	name   string
+	logger log.Logger
+	client *alioss.Client
+	config Config
+	bucket *alioss.Bucket
+}
+
+func NewTestBucket(t testing.TB, location string) (objstore.Bucket, func(), error) {
+	c := configFromEnv()
+	if err := ValidateForTests(c); err != nil {
+		return nil, nil, err
+	}
+	if c.Bucket != "" && os.Getenv("THANOS_ALLOW_EXISTING_BUCKET_USE") == "" {
+		return nil, nil, errors.New("OSS_BUCKET is defined. Normally this tests will create temporary bucket " +
+			"and delete it after test. Unset OSS_BUCKET env variable to use default logic. If you really want to run " +
+			"tests against provided (NOT USED!) bucket, set THANOS_ALLOW_EXISTING_BUCKET_USE=true. WARNING: That bucket " +
+			"needs to be manually cleared. This means that it is only useful to run one test in a time. This is due " +
+			"to safety (accidentally pointing prod bucket for test) as well as aliyun oss not being fully strong consistent.")
```

Comment:
This documentation is wrong. Alibaba OSS is strong consistent!

Reply:
@wujinhu  OK ， I will correct it